roachtest: add -c/--cluster for running a test on an existing cluster

petermattis opened the pull request:
Add `-c/--cluster` flag for running a test or tests on an existing
cluster. When running on an existing cluster, the cluster is wiped
before and after the test, but no attempt is made to create or destroy
it. Like running a test locally, running on an existing cluster forces
parallelism to 1.

Fixes #22644

Release note: None

petermattis committed:
roachtest: add -c/--cluster for running a test on an existing cluster

Add `-c/--cluster` flag for running a test or tests on an existing
cluster. When running on an existing cluster, the cluster is wiped
before and after the test, but no attempt is made to create or destroy
it. Like running a test locally, running on an existing cluster forces
parallelism to 1.

Fixes #22644

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/22795)
<!-- Reviewable:end -->

petermattis committed:
roachtest: use local SSDs for kv tests

Scale the concurrency of the test by the number of nodes in the cluster.

Release note: None
